plugins/com.google.cloud.tools.eclipse.login/src/com/google/cloud/tools/eclipse/login/ui/AccountsPanel.java

```diff
@@ -81,45 +82,67 @@ protected Control createDialogArea(Composite parent) {
-      int avatarSize = emailSize.y * 3;
-
-      GridDataFactory.swtDefaults().hint(avatarSize, avatarSize).applyTo(avatar);
-      GridLayoutFactory.fillDefaults().numColumns(2).applyTo(accountRow);
-      GridLayoutFactory.fillDefaults().generateLayout(secondColumn);
-
-      if (account.getName() != null) {
-        name.setText(account.getName());
-      }
-      email.setText(account.getEmail());  // email is never null.
-
-      if (account.getAvatarUrl() != null) {
-        try {
-          String url = resizedImageUrl(account.getAvatarUrl(), avatarSize);
-          imageLoader.loadImage(url, avatar);
-        } catch (MalformedURLException ex) {
-          logger.log(Level.WARNING, "malformed avatar image URL", ex);
-        }
+    Account account = getAccount();
+    Composite accountRow = new Composite(accountArea, SWT.NONE);
+    Label avatar = new Label(accountRow, SWT.NONE);
+    Composite secondColumn = new Composite(accountRow, SWT.NONE);
+    Label name = new Label(secondColumn, SWT.LEAD);
+    Label email = new Label(secondColumn, SWT.LEAD);
+    Label separator = new Label(accountArea, SWT.HORIZONTAL | SWT.SEPARATOR);
+    separator.setLayoutData(new GridData(SWT.FILL, SWT.BEGINNING, true, false));
+
+    // <Avatar size> = 3 * <email label height>
+    Point emailSize = email.computeSize(SWT.DEFAULT, SWT.DEFAULT);
+    int avatarSize = emailSize.y * 3;
+
+    GridDataFactory.swtDefaults().hint(avatarSize, avatarSize).applyTo(avatar);
+    GridLayoutFactory.fillDefaults().numColumns(2).applyTo(accountRow);
+    GridLayoutFactory.fillDefaults().generateLayout(secondColumn);
+
+    account.getName().ifPresent(name::setText);
+    email.setText(account.getEmail());  // email is never null.
+
+    if (account.getAvatarUrl().isPresent()) {
+      try {
+        String url = resizedImageUrl(account.getAvatarUrl().get(), avatarSize);
+        imageLoader.loadImage(url, avatar);
+      } catch (MalformedURLException ex) {
+        logger.log(Level.WARNING, "malformed avatar image URL", ex);
       }
     }
   }
+  
+  @VisibleForTesting
+  void createNonLoggedInAccountsPane(Composite accountArea) {
+    Composite messageRow = new Composite(accountArea, SWT.NONE);
+    Label message = new Label(messageRow, SWT.NONE);
+    message.setText(Messages.getString("ACCOUNTS_PANEL_NO_ADC_DETECTED_MESSAGE"));
+    Composite linkRow = new Composite(accountArea, SWT.NONE);
+    Link helpLink = new Link(linkRow, SWT.NONE);
+    helpLink.setText(Messages.getString("ACCOUNTS_PANEL_NO_ADC_DETECTED_LINK"));
+    helpLink.addSelectionListener(new OpenUriSelectionListener(new ErrorDialogErrorHandler(accountArea.getShell())));
+    GridLayoutFactory.fillDefaults().numColumns(1).applyTo(messageRow);
+    GridLayoutFactory.fillDefaults().generateLayout(linkRow);
+    
+  }
+  
+  private Account getAccount() {
+    try {
+      return apiFactory.getAccount();
+    } catch (IOException ex) {
+      return null;
```

Comment:
If this returns `null`, then the above `account.getName()` and `account.getEmail()` will throw a NullPointerException.